src/xopen/__init__.py

```diff
@@ -650,6 +652,50 @@ def __init__(
         )
 
 
+class PipedXzReader(PipedCompressionReader):
+    """
+    Open a pipe to xz for reading an xz-compressed file.
+    """
+
+    # TODO: implement threaded decompression once XZ releases it
+    def __init__(
+        self, path, mode: str = "r", *, encoding="utf-8", errors=None, newline=None
+    ):
+        super().__init__(
+            path, ["xz"], mode, encoding=encoding, errors=errors, newline=newline
+        )
+
+
+class PipedXzWriter(PipedCompressionWriter):
+    """
+    Write xz-compressed files by running an external xz process and
+    piping into it. xz can compress using multiple cores.
+    """
+
+    def __init__(
+        self,
+        path,
+        mode: str = "wt",
+        threads: Optional[int] = None,
+        *,
+        encoding="utf-8",
+        errors=None,
+        newline=None,
+    ):
+        # Use default compression level for xz: 6
+        super().__init__(
+            path,
+            ["xz"],
+            mode,
+            6,
```

Comment:
Please pass trough the compresslevel as highlighted above. A default of `None` will automatically set no flag and let xz choose the default compression level.